### PR TITLE
big commit, linting, logging and vert_coord

### PR DESCRIPTION
**Pull request by oloapinivad**
So this pull request introduces 

- Better structured logger that is defined in each of the required module as it is reccomended to be done. This should solve also one of the bug we had when the cdo generation crash and we do not get any error message. 
- A bit of litinting with `autopep8`
- An updated feature for `find_vert_coord()`, with some new defaults and the possibility of detecting more than one vert_coord (but raising an error)

The only things I am not super happy with is the fact there is no simple way to modify the logging at the MODULE level, and this has to be changed manually in the files. This is mostly because we do not have an extensive class but rather a collection of function, and I did not want to play with the root logger to avoid messing with other tools.

### CODE REVIEW

**Comment by oloapinivad** [comment]
@jhardenberg when you have time please have a look if you are happy with that... we could ship a new pip version as soon as this is merged...

**Review by jhardenberg — COMMENTED** [review]
looks good. Just a very few minor comments

File: smmregrid/__init__.py [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 from .regrid import Regridder, regrid
 from .cdo_weights import cdo_generate_weights
-from .util import find_vert_coord
+#from .util import find_vert_coords
```
[the remark this answers is not in the document]
Comment (oloapinivad) [reply smmregrid/__init__.py]: done!

File: smmregrid/regrid.py [hunk 2]
```diff
@@ -72,10 +73,10 @@ def apply_weights(source_data, weights, weights_matrix=None, masked=True, space_
 
         # we keep time bounds, and we ignore all the rest
         if 'time' in source_data.name:
-            logging.info('original %s', source_data.name)
+            loggy.info('original %s', source_data.name)
```
[the remark this answers is not in the document]
Comment (oloapinivad) [reply smmregrid/regrid.py:76]: logging is the package, I need another name. logger, loggy, logos, whatever :-)

File: smmregrid/util.py [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
 
     if isinstance(xfield, str):
         xfield = xarray.open_dataset(xfield)
-    for coord in default_vert_coords:
-        if coord in xfield.coords:
-            return coord
\ No newline at end of file
+
+    vcoords = list(set(xfield.coords.keys()).intersection(default_vert_coords))
+    if len(vcoords) > 1:
```
[the remark this answers is not in the document]
Comment (oloapinivad) [reply smmregrid/util.py:21]: I will check our implementation in aqua, in principle we specify the `vert_coord` there. We can merge this and test it from the main without updating the pypi tool...

**Comment by oloapinivad** [comment]
Fixes implemented, if I have time I will double check the logging module implementation since there might be a better option...

**Comment by oloapinivad** [comment]
I imported the logging structure from Ecmean which is more robust. Did a few changes to the enviornment file and the workflow so that is more updated. Tests are passing, and the `vert_coord` safe check should not conflict with our other project which takes care of multiple vertical coordinates itself.

**Comment by oloapinivad** [comment]
Merging this to the main

**Pull request merged**